.github/workflows/build.yaml

```diff
@@ -4,41 +4,144 @@ on:
+            version: "10"
+
+          - os: ubuntu-16.04
+            compiler: clang
+            version: "6.0"
+
+          - os: ubuntu-18.04
+            compiler: clang
+            version: "7"
+
+          - os: ubuntu-18.04
+            compiler: clang
+            version: "8"
+
+          - os: ubuntu-20.04
+            compiler: clang
+            version: "9"
 
+          - os: ubuntu-20.04
+            compiler: clang
+            version: "10"
+
+          - os: macOS-latest
+            compiler: xcode
+            version: "11"
+
+          - os: macOS-latest
+            compiler: xcode
+            version: "11.3"
     steps:
-      - name: Get source
-        uses: actions/checkout@v2
+      - name: Install dependencies (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          if [ "${{ matrix.config.os }}" = "ubuntu-20.04" ]; then
+            sudo apt-get install -y ninja-build elfutils libzstd-dev
+          else
+            sudo apt-get install -y ninja-build elfutils libzstd1-dev
+          fi
+
+          if [ "${{ matrix.config.compiler }}" = "gcc" ]; then
+            echo "CC=gcc-${{ matrix.config.version }}" >> $GITHUB_ENV
+            echo "CXX=g++-${{ matrix.config.version }}" >> $GITHUB_ENV
+
+            # Disable warnings on ancient compiler.
+            if [ "${{ matrix.config.version }}" = "4.8" ]; then
+              echo "CMAKE_PARAMS=-DWARNINGS_AS_ERRORS=OFF" >> $GITHUB_ENV
+            fi
+
+            sudo apt install -y g++-${{ matrix.config.version }} g++-${{ matrix.config.version }}-multilib
+          else
+            if [ "${{ matrix.config.version }}" = "3.4" ]; then
+              echo "CC=clang" >> $GITHUB_ENV
+              echo "CXX=clang++" >> $GITHUB_ENV
+              echo "CXXFLAGS=-stdlib=libc++" >> $GITHUB_ENV
 
-      - name: Install dependencies (Ubuntu 16 & 18)
-        if: startsWith(matrix.os, 'ubuntu') && matrix.os != 'ubuntu-20.04'
-        run: sudo apt-get install elfutils libzstd1-dev
+              # clang-3.4 is available from trusty
+              sudo add-apt-repository "deb http://dk.archive.ubuntu.com/ubuntu/ trusty main"
+              sudo add-apt-repository "deb http://dk.archive.ubuntu.com/ubuntu/ trusty universe"
+  
+              # Disable warnings on ancient compiler.
+              # doctest does not compile with clang-3.4.
+              echo "CMAKE_PARAMS=-DWARNINGS_AS_ERRORS=OFF -DENABLE_TESTING=OFF" >> $GITHUB_ENV
+            else
+              echo "CC=clang-${{ matrix.config.version }}" >> $GITHUB_ENV
+              echo "CXX=clang++-${{ matrix.config.version }}" >> $GITHUB_ENV
 
-      - name: Install dependencies (Ubuntu 20)
-        if: matrix.os == 'ubuntu-20.04'
-        run: sudo apt-get install elfutils libzstd-dev
+              if [ "${{ matrix.config.version }}" = "3.5" ]; then
+                # Disable warnings on ancient compiler.
+                echo "CMAKE_PARAMS=-DWARNINGS_AS_ERRORS=OFF" >> $GITHUB_ENV
+              fi
```

Comment:
Should this be removed as well or do you plan to reintroduce a test for Clang 3.5?

Reply:
I want to support clang 3.5 - 5 since all it requires is #693, that sounds reasonable.
However I moved this setting to cmake, since it's not specific to CI.